mlrun/serving/v2_serving.py

```diff
@@ -225,13 +230,29 @@ def _pre_event_processing_actions(self, event, event_body, op):
         request = self.preprocess(event_body, op)
         return self.validate(request, op)
 
+    @property
+    def versioned_model_name(self):
+        version = self.version or "latest"
+        return f"{self.name}:{version}"
+
     def do_event(self, event, *args, **kwargs):
         """main model event handler method"""
         start = now_date()
         original_body = event.body
         event_body = _extract_input_data(self._input_path, event.body)
         event_id = event.id
         op = event.path.strip("/")
+
+        # Generating model endpoint ID based on function uri and model version
+        model_endpoint_uid = mlrun.common.model_monitoring.create_model_endpoint_uid(
+            function_uri=self.context.server.function_uri,
+            versioned_model=self.versioned_model_name,
+        ).uid
```

Comment:
It's not always initialized, e.g. when running a mock server. And anyway, we're planning to move `_init_endpoint_record` from here to the deployment phase.

Reply:
If we rely on `post_init` here, then `self.model_endpoint_uid` will be `None` in mock server. I don't remember the specific failure that causes, but it did fail from what I remember.

Again, since our intention is to move `_init_endpoint_record` to the deployment stage ([ML-7930](https://iguazio.atlassian.net/browse/ML-7930)), I can't imagine why we would want to spend time on this, only to change the code back shortly after.

Reply:
I remembered that there was an issue with this, but it works fine when I try it now, so I pushed this change. Do note we will sort of have to change it back soon as part of [ML-7930](https://iguazio.atlassian.net/browse/ML-7930).